### PR TITLE
Flutter/feature 198

### LINKED ISSUE #198: As a guardian I would like to be able to delete week plans so that I can avoid clutter and not be overwhelmed

### PR DESCRIPTION
**Pull request by boisenme**
Closes #198 

ezgif-1-0b9125626827

### CODE REVIEW

**Comment by User12837985123** [comment]
@aau-giraf/sw602f19 you have been assigned this. Here's your checklist. I marked @Jacob6565 as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by User12837985123** [comment]
@aau-giraf/sw612f19 you have been assigned this. Here's your checklist. I marked @thomaslhansen as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [x] Is the code in the right place? (Both in terms of folder structure and class structure)
- [x] Could this code, to the best of your knowledge, have reused existing code?
- [x] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [x] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [x] Are names meaningful and self-documenting?
- [x] Is it understandable, by reading the code, what it does?
- [x] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [x] Is it understandable, by reading the tests, what they do?
- [x] Do the tests cover sensible cases? Both happy paths and exception paths?
- [x] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [x] Are error messages in the code understandable?
- [x] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [x] Are the features that the code claims to implement, actually implemented?
- [x] Does it look like the code has bugs?
- [x] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [x] Are UI messages, that the user might meet, actually helpful?
- [x] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [x] Is there well written code in this? Remember to tell the developer!

**Review by B97theFallen — DISMISSED** [review]

File: test/blocs/weekplans_bloc_test.dart [hunk 1]
```diff
@@ -67,4 +71,124 @@ void main() {
     bloc.weekNameModels.listen((_) {}, onDone: done);
     bloc.dispose();
   }));
+
+  test('Adds a weekmodel to a list of marked weekmodels', async((DoneFn done) {
```
Comment (User12837985123) [inline test/blocs/weekplans_bloc_test.dart:75]: Måske overser jeg noget, men kan ikke helt se at der bliver tilføjet/markeret en marked weekmodel. Det ligner blot at i tester at der er 1 marked weekmodel, men kan ikke lige se hvor i angiver den som marked.

**Review by B97theFallen — DISMISSED** [review]

File: test/blocs/weekplans_bloc_test.dart [hunk 2]
```diff
@@ -67,4 +71,124 @@ void main() {
...
+
+    bloc.markedWeekModels
+        .skip(1)
+        .listen((List<WeekModel> markedWeekModelsList) {
+      expect(markedWeekModelsList.length, 0);
+      done();
+    });
+
+    // Clears the list of marked weekmodels.
+    bloc.clearMarkedWeekModels();
```
Comment (User12837985123) [inline test/blocs/weekplans_bloc_test.dart:117]: Burde man ikke først tjekke at markedWeekModelsList.length == 0 efter at have kaldt bloc.clearMarkedWeekModels?

**Review by User12837985123 — CHANGES_REQUESTED** [review]

File: lib/blocs/weekplans_bloc.dart [hunk 3]
```diff
@@ -78,6 +78,12 @@ class WeekplansBloc extends BlocBase {
           .get(_user.id, weekNameModel.weekYear, weekNameModel.weekNumber)
           .listen((WeekModel results) {
         weekModels.add(results);
+        weekModels.sort((WeekModel a, WeekModel b) {
+          if (a.name == 'Tilføj ugeplan'){
+            return -1;
+          }
+          return a.name.compareTo(b.name);
```
Comment (B97theFallen) [inline lib/blocs/weekplans_bloc.dart]: Ser ud som om i sorterer efter navnet, men det virker ikke til at den gør det i min emulator.
De er dog konsistente med hvor de placeres i forhold til hinanden.

Jeg har pullet siden sidste commit (commit 2c06493b783a0a85fd0733d206f154091bd54d74)

Comment (LukasRH) [reply lib/blocs/weekplans_bloc.dart]: Ideen med det var også bare at få dem til at stå på den samme måde hver gang.
Så der er ikke tænkt over en konkret sotering, det er nok et andet issue.

Comment (B97theFallen) [reply lib/blocs/weekplans_bloc.dart]: Okay, det er fint så.

Comment (boisenme) [reply lib/blocs/weekplans_bloc.dart]: Vi fikser det :D

File: test/blocs/weekplans_bloc_test.dart (hunk 2 above)
Comment (llundbo) [reply test/blocs/weekplans_bloc_test.dart:117]: Nej det det er faktisk meningen, at vi lytter hvor vi gør. Listen virker nemlig sådan, at man lytter gennem hele testen, dvs. at efter bloc.clarMarkedWeekModels bliver kaldt så tjekkes der om længden er 0 oppe i Listen :-).

File: test/blocs/weekplans_bloc_test.dart (hunk 1 above)
Comment (llundbo) [reply test/blocs/weekplans_bloc_test.dart:75]: Der bliver tilføjet en markeret WeekModel på linje 84. Listen fungerer på samme måde som jeg beskrev i min anden comment, så det skulle være i orden.

**Review by B97theFallen — APPROVED** [review]

**Review by User12837985123 — APPROVED** [review]

**Review by andpaacoke — APPROVED** [review]
PO approved 👍

**Pull request merged**